liferay-frontend/liferay-portal
# LPS-153324 Add System Objects relationship endpoints

Related ticket -> [LPS-153324](https://issues.liferay.com/browse/LPS-153324)
___
### What do we do?
This is the first PR to create the relationship endpoints of the system objects in the existing APIs. Let's see an example.
Imagine a University with their subjects and their students in this way:
* University is a custom object. Has a one-to-many relationship with Student and a many-to-many relationship with Subject
* A user of Liferay represents a student, so, it is a System Object. It also has a many-to-many relationship with Subject
* Subject is a custom object.
According to these requisites, there should be an endpoint to retrieve the student's subject, and this endpoint should be in the existing REST API `headless-admin-user`

This PR adds that endpoint and its logic to be able to retrieve the related objects of a system object.
IMPORTANT: These only expose the endpoint, but this will not appear in the API Explorer.

### How to test it
Deploy the following modules:
* object-api
* object-service
* object-rest-api
* object-rest-impl

Set the following feature flag to true:
* `feature.flag.LPS-153324`

Create the necessary objects and relationships according to the previous example and populate them with some data.

Make a request to get the student's subjects, for example:
```
curl "http://localhost:8080/o/headless-admin-user/v1.0/user-accounts/41492/studentSubjects?fields=subjectName"
```
This request should return something like:
```
{
  "actions": {},
  "facets": [],
  "items": [
    {
      "subjectName": "Pyshics"
    },
    {
      "subjectName": "Calculus"
    },
    {
      "subjectName": "Programming"
    },
    {
      "subjectName": "Algorithms"
    }
  ],
  "lastPage": 1,
  "page": 1,
  "pageSize": 4,
  "totalCount": 4
}
```

This is just the first PR to add support to the different relationships in Object (custom and system). The next steps are:
* Add the previous endpoints to the API Explorer
* Use this logic in the Custom Objects so there will be compatibility between custom and system object relationships
* Add the put endpoint to create relationships between objects (custom and system) and add it to the API Explorer
